controllers/searchFriends.js

```diff
@@ -1,52 +1,73 @@
 const { getMongoClient } = require('../utils/database');
 
-const searchFriends = (async (req, res, next) => 
-{
+const searchFriends = async (req, res, next) => {
     // incoming: userId, search
 
     // outgoing: results or error
 
     let _search = '';
-    try
-    {
-        const {search } = req.body;
+    let ret = {};
+    ret.error = '';
+
+    try {
+        const { search } = req.body;
         _search = search.trim();
-    }    
-    catch (e)
-    {
+        if (_search === '') throw new Error();
+    } catch (e) {
         res.locals.ret.error = 'Bad request syntax. Missing or incorrect information.'
         res.status(400).json(res.locals.ret);
         return;
     }
 
-    const requestBody = 
-    {
-        user1: res.locals.token.login,
-        user2: _search
-    }
-    try{
+    try {
         const client = getMongoClient();
-        client.connect();
+        await client.connect();
         const db = client.db();
 
-        const results = await db.collection('Relationships').find(requestBody).toArray;
-        
-        if (results.length > 0) {
-            res.status(200).json({ results: results });
-        } 
-        else {
-            res.locals.ret.error = 'Could not find friendship.'
-            res.status(200).json(res.locals.ret);
+        // check if search exists as a user by first name, last name, email, or username (case-insensitive)
+        const users = await db.collection('Users').find({
+            $or: [
```

Comment:
I have not ran it so I am not sure if this works or not, but does this do partial matches? Or does it have to be an exact match to one of the user's fields?